### ISSUE TITLE
Android won't use upside-down portrait mode

### ISSUE THREAD

**Issue by meepingsnesroms** [libretro/RetroArch#4464]
This may seem trivial but when you have a charger plugged in and the port is on the bottom of your phone it is quite useful.

Should be very simple to fix by just adding the value to the apps .xml file.

**Comment by sergiobenrocha2** [comment]
You'll be unable to play some wonder swan games:

![screenshot_2017-01-20-21-25-52___ 01](https://cloud.githubusercontent.com/assets/5431285/22168922/75c7f9c2-df57-11e6-9713-35447eafaf8b.png)

![screenshot_2017-01-20-21-33-22 01](https://cloud.githubusercontent.com/assets/5431285/22169040/485d0f9e-df58-11e6-9f9a-c8a6cbe94c42.png)

**Comment by meepingsnesroms** [comment]
You still can, settings>video>rotation = fixed landscape.

**Comment by sergiobenrocha2** [comment]
There's no "fixed landscape" option. Try play Judgment Silverword with gamepad/flat/wonderswan.cfg overlay, you'll understand.

**Comment by meepingsnesroms** [comment]
Than lock rotation needs to be added with it.(will the os lock orientation toggle not work?)

**Comment by fr500** [comment]
uhh... if it allowed that rotation it would just work... it rotates the
whole viewport, overlays included. IE, it will look the same in both
orientations so if one works the other one does too.

**Comment by sergiobenrocha2** [comment]
Dudes...

![screenshot_2017-01-21-17-19-31](https://cloud.githubusercontent.com/assets/5431285/22177075/4be24c46-dffe-11e6-8cfb-c954825378c4.png)

![screenshot_2017-01-21-17-21-11](https://cloud.githubusercontent.com/assets/5431285/22177080/552b942e-dffe-11e6-87fc-38e594754edd.png)

Rotate = 90 degree

Dpad is the Y buttons, shoot is X1 and X2

**Comment by fr500** [comment]
He's not talking about rotating within RetroArch, just to honor device rotation

**Comment by sergiobenrocha2** [comment]
If you honor the device rotation in upside-down, you'll be unable to play properly. I think I'll need to make a video to describe it heh

**Comment by fr500** [comment]
![](https://cloud.githubusercontent.com/assets/5431285/22177075/4be24c46-dffe-11e6-8cfb-c954825378c4.png)

It would look exactly like this

**Comment by sergiobenrocha2** [comment]
Yeah, try play now. Hint: X buttons are not the d-pad in this game

**Comment by fr500** [comment]
yes but that's not really a problem that concerns this issue

**Comment by meepingsnesroms** [comment]
Then we just need a portrait wonderswan overlay.

**Comment by sergiobenrocha2** [comment]
Make it optional at least. Don't over complicate things more than which is already in RA

**Comment by fr500** [comment]
dude, this will literally cause nothing unexpected it's just allowing one extra rotation, it will work exactly as it does now.
It's just one change to the manifest to allow rotating the device upside down.

**Comment by sergiobenrocha2** [comment]
It does depends on not rotating when in upside-down position, this is the main point. Not rotating in upside-down is a feature, not a "bug", since you can't force landscape position in RA without an external app.

It'll affect Linx games too (klax).

**Comment by fr500** [comment]
but the only thing you have to do is.. don't rotate the screen upside down...

**Comment by andoalon** [comment]
My two cents: as said by others, playing games in portrait mode while having a charger or USB controller plugged in is uncomfortable or impossible.

I'm not familiar with those games that people say there are going to be problems with, but if that's the case, there's an easy solution: just lock the rotation of the device (in Android no RetroArch). I can't think of any case that isn't solved by that (do let me know if you can think of an example).

However, for my case (wanting to play upside down) the only workaround is to fix the video rotation to 180º through the advanced video settings, which then causes all the on-screen controls as well as any overlays to be upside down (including retroachievement pop-ups, etc.). For me, this is pretty clear-cut
